dvc/exceptions.py

```diff
@@ -260,9 +260,17 @@ def __init__(self, etag, cached_etag):
 
 
 class OutputFileMissingError(DvcException):
-    def __init__(self, path):
+    def __init__(self, path, cause):
         super(OutputFileMissingError, self).__init__(
-            "Can't find {} neither locally nor on remote".format(path)
+            "Can't find '{}' neither locally nor on remote".format(path),
+            cause=cause,
+        )
+
+
+class NotCachedOutputFileMissingError(DvcException):
+    def __init__(self, path_info):
+        super(NotCachedOutputFileMissingError, self).__init__(
+            "Can't find '{}'. This file was not cached.".format(path_info)
         )
```

Comment:
Both class name and message blur more than they explain. And, BTW, what's the value of distinguishing this from `OutputFileMissing`?

I say we should stop inventing more exception classes. This complicates both our code and API.